src/control-service.vala

```diff
@@ -383,6 +393,34 @@ namespace Frida {
 				channel.spawn_removed (info);
 		}
 
+		private void notify_output (uint pid, int fd, uint8[] data) {
+			all_control_channels ().foreach (channel => {
+				channel.output (pid, fd, data);
+				return true;
+			});
+		}
```

Comment:
These should ideally be in the same order as the signal definitions in the `HostSession` interface.